jujuclient/mem.go

```diff
@@ -13,6 +15,8 @@ import (
 
 // MemStore is an in-memory implementation of ClientStore.
 type MemStore struct {
+	mu sync.Mutex
```

Comment:
While I'm mostly OK with adding a mutex here, the MemStore wasn't necessarily meant to be goroutine safe. Are the tests doing inappropriate things with it? Can you please link to the race that this fixes?